### PR TITLE
Swift snippet using the authCallback.

### PR DESCRIPTION
**Pull request by EvgenyKarkan**
#461 
For preliminary review, if everything is OK I'll provide an `ObjectiveC` snippet.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -304,6 +304,33 @@ channel.presence.history { presencePage, error in
...
+Assume given auth `json` data, then `ARTClientOptions` and `ARTRealtime` objects can be instantiated as follow:
+
+**Swift**
+
+```swift
+let clientOptions = ARTClientOptions()
+clientOptions.authCallback = { params, callback in
+    let tokenParams = ARTTokenParams(clientId: json["clientId"] /*string*/)
+
+    let tokenRequest = ARTTokenRequest(tokenParams: tokenParams),
```
Comment (tcard) [inline README.md]: Incorrect closing param there.

File: README.md [hunk 2]
```diff
@@ -304,6 +304,33 @@ channel.presence.history { presencePage, error in
...
+                                           keyName: json["keyName"], /*string*/
```
Comment (tcard) [inline README.md]: The `/*string*/` comments are too noisy in my opinion. Swift is already strongly typed.

File: README.md [hunk 3]
```diff
@@ -304,6 +304,33 @@ channel.presence.history { presencePage, error in
...
+
+A callback to call to obtain a signed token request.
+Assume given auth `json` data, then `ARTClientOptions` and `ARTRealtime` objects can be instantiated as follow:
+
+**Swift**
+
+```swift
+let clientOptions = ARTClientOptions()
+clientOptions.authCallback = { params, callback in
+    let tokenParams = ARTTokenParams(clientId: json["clientId"] /*string*/)
```
Comment (tcard) [inline README.md]: Probably worth it simulating getting this from somewhere:

```swift
clientOptions.authCallback = { params, callback in
    getTokenRequestJSONFromYourServer(params) { json, err in
        // handle error
        let tokenParams = ...
    }
}
```

**Comment by EvgenyKarkan** [comment]
@tcard thanks for feedback, I will improve.

**Comment by EvgenyKarkan** [comment]
Please find time to review @tcard

**Comment by tcard** [comment]
LGTM

File: README.md [hunk 4]
```diff
@@ -334,6 +334,31 @@ clientOptions.authCallback = { params, callback in
 let client = ARTRealtime(options: clientOptions)
 ```
 
+**Objective-C**
+
+```objective-c
+ARTClientOptions *clientOptions = [[ARTClientOptions alloc] init];
+    clientOptions.authCallback = ^(ARTTokenParams *params, void(^callBack)(id<ARTTokenDetailsCompatible>, NSError*)) {
```
Comment (tcard) [inline README.md]: From this line on there's one extra level of indentation.

File: README.md [hunk 5]
```diff
@@ -334,6 +334,31 @@ clientOptions.authCallback = { params, callback in
...
+                                                                                   nonce: json[@"nonce"]
+                                                                                     mac: json[@"mac"]];
+            tokenRequest.clientId = json[@"clientId"];
+            tokenRequest.ttl = [json[@"ttl"] doubleValue];
+            tokenRequest.capability = json[@"capability"];
+            tokenRequest.timestamp = [NSDate dateWithTimeIntervalSince1970: [json[@"timestamp"] doubleValue] / 1000];
+
+            callBack(tokenRequest, nil);
+        }];
+    };
```
Comment (tcard) [inline README.md]: I think the closing param should be here, shouldn't it?

File: README.md [hunk 6]
```diff
@@ -334,6 +334,31 @@ clientOptions.authCallback = { params, callback in
...
+            ARTTokenRequest *tokenRequest = [[ARTTokenRequest alloc] initWithTokenParams: tokenParams
+                                                                                 keyName: json[@"keyName"]
+                                                                                   nonce: json[@"nonce"]
+                                                                                     mac: json[@"mac"]];
+            tokenRequest.clientId = json[@"clientId"];
+            tokenRequest.ttl = [json[@"ttl"] doubleValue];
+            tokenRequest.capability = json[@"capability"];
+            tokenRequest.timestamp = [NSDate dateWithTimeIntervalSince1970: [json[@"timestamp"] doubleValue] / 1000];
+
+            callBack(tokenRequest, nil);
```
Comment (tcard) [inline README.md]: It's called `callback` (all lower case) in all other examples.

File: README.md [hunk 7]
```diff
@@ -334,6 +334,31 @@ clientOptions.authCallback = { params, callback in
...
+            ARTTokenParams *tokenParams = [[ARTTokenParams alloc] initWithClientId: json[@"clientId"]];
+
+            ARTTokenRequest *tokenRequest = [[ARTTokenRequest alloc] initWithTokenParams: tokenParams
+                                                                                 keyName: json[@"keyName"]
+                                                                                   nonce: json[@"nonce"]
+                                                                                     mac: json[@"mac"]];
+            tokenRequest.clientId = json[@"clientId"];
+            tokenRequest.ttl = [json[@"ttl"] doubleValue];
+            tokenRequest.capability = json[@"capability"];
+            tokenRequest.timestamp = [NSDate dateWithTimeIntervalSince1970: [json[@"timestamp"] doubleValue] / 1000];
```
Comment (tcard) [inline README.md]: We don't put a space between the `:` and the argument in other examples.

**Comment by tcard** [comment]
A few stylistic comments but LGTM otherwise.

File: README.md (hunk 5 above)
Comment (EvgenyKarkan) [inline README.md]: Sorry, but Xcode shows no warnings here, seems ok to me, what do you mean?

Comment (tcard) [inline README.md]: Sorry, you're right, I parsed this wrong in my head.

**Comment by EvgenyKarkan** [comment]
Ok, thanks @tcard , please review with stylistic improvements.

**Comment by tcard** [comment]
LGTM

**Comment by EvgenyKarkan** [comment]
Ok, as far as I understand - I can merge PR after successful review, right @tcard ?

**Comment by tcard** [comment]
That's right @EvgenyKarkan.

**Pull request merged**